File: code/java/src/main/java/bridge/RemoteControl.java
package bridge;

abstract class RemoteControl implements  Device{
    protected Device device;

    public RemoteControl(Device device) {
        this.device = device;
    }

    @Override
    public void powerOn() {
        device.powerOn();
    }

    @Override
    public void powerOff() {
        device.powerOff();
    }

    @Override
    public void adjustVolume(int volume) {
        device.adjustVolume(volume);
    }

    @Override
    public void changeChannel(int channel) {
        device.changeChannel(channel);
    }
}

File: code/typescript/src/bridge/RemoteControl.ts
import { Device } from './Device';

export abstract class RemoteControl implements Device {
    protected device: Device;

    constructor(device: Device) {
        this.device = device;
    }

    powerOn(): void {
        this.device.powerOn();
    }

    powerOff(): void {
        this.device.powerOff();
    }

    adjustVolume(volume: number): void {
        this.device.adjustVolume(volume);
    }

    changeChannel(channel: number): void {
        this.device.changeChannel(channel);
    }
}
